tests/drivers/aio/aio_basic_api/src/test_callback.c

```diff
@@ -1,54 +1,149 @@
+#define PIN_IN 10	/* GPIO_SS_AIN_10 */
+#elif defined(CONFIG_BOARD_QUARK_SE_C1000_DEVBOARD_SS)
+#define PIN_OUT 3	/* GPIO_SS_3 */
+#define PIN_IN 10	/* GPIO_SS_AIN_10 */
+#elif defined(CONFIG_BOARD_QUARK_D2000_CRB)
+#define PIN_OUT 8	/* GPIO_8 */
+#define PIN_IN 10	/* AIN_10 */
+#elif defined(CONFIG_BOARD_ARDUINO_101)
+#define PIN_OUT 16	/* GPIO_16 */
+#define PIN_IN 10	/* AIN_10 */
+#elif defined(CONFIG_BOARD_ARDUINO_101_SSS)
+#define PIN_OUT 3	/* GPIO_SS_3 */
+#define PIN_IN 10	/* AIN_10 */
+#endif
 
 static int cb_cnt;
+static struct device *aio_dev;
+static struct device *gpio_dev;
 
 static void callback(void *param)
 {
+	int polarity = *(int *)param;
+
 	cb_cnt++;
+
+	TC_PRINT("*** callback triggered %s\n",
+		(polarity == AIO_CMP_POL_RISE) ? "rising" : "falling"
+	);
+
+	if (polarity == AIO_CMP_POL_FALL) {
+		gpio_pin_write(gpio_dev, PIN_OUT, 1);
+	} else {
+		gpio_pin_write(gpio_dev, PIN_OUT, 0);
+	}
+
+	if (aio_cmp_get_pending_int(aio_dev)) {
+		TC_PRINT("Catch aio_cmp pending interrupt\n");
+	} else {
+		TC_PRINT("Fail to catch aio_cmp pending interrupt\n");
+	}
+
+	aio_cmp_disable(aio_dev, PIN_IN);
 }
 
-static void set_aio_callback(int polarity, int disable)
+static int set_aio_callback(int polarity, int disable)
 {
-	struct device *aio_dev = device_get_binding(AIO_CMP_DEV_NAME);
+	struct device *pinmux = device_get_binding(PINMUX_NAME);
 
-	struct device *gpio_dev = device_get_binding(GPIO_DEV_NAME);
+	if (!pinmux) {
+		TC_PRINT("Cannot get PINMUX\n");
+		return TC_FAIL;
+	}
 
-	gpio_pin_configure(gpio_dev, PIN_OUT, GPIO_DIR_OUT);
+	aio_dev = device_get_binding(AIO_CMP_DEV_NAME);
+	if (!aio_dev) {
+		TC_PRINT("AIO Device binding failed\n");
+		return TC_FAIL;
+	}
+
+	gpio_dev = device_get_binding(GPIO_DEV_NAME);
+	if (!gpio_dev) {
+		TC_PRINT("GPIO Device binding failed\n");
+		return TC_FAIL;
+	}
+
+	if (gpio_pin_configure(gpio_dev, PIN_OUT, GPIO_DIR_OUT)) {
+		TC_PRINT("Fail to configure GPIO Pin %d\n", PIN_OUT);
+		return TC_FAIL;
+	}
+
+	if (pinmux_pin_set(pinmux, PIN_IN, PINMUX_FUNC_B)) {
+		TC_PRINT("Fail to set pin func, %u\n", PIN_IN);
+		return TC_FAIL;
+	}
+
+	if (polarity == AIO_CMP_POL_FALL) {
+		gpio_pin_write(gpio_dev, PIN_OUT, 1);
+	} else {
+		gpio_pin_write(gpio_dev, PIN_OUT, 0);
+	}
 
 	/* config AIN callback */
 	zassert_true(aio_cmp_configure(aio_dev, PIN_IN,
 				      polarity, AIO_CMP_REF_A,
 				      callback, (void *)aio_dev) == 0,
 		    "ERROR registering callback");
 	if (disable == 1) {
-		aio_cmp_disable(aio_dev, PIN_IN);
+		if (aio_cmp_disable(aio_dev, PIN_IN)) {
+			TC_PRINT("Fail to register callback\n");
+			return TC_FAIL;
+		}
 	}
 
-	/* update the AIN input to trigger state */
-	gpio_pin_write(gpio_dev, PIN_OUT,
-		       (polarity == AIO_CMP_POL_RISE) ? 0 : 1);
 	k_sleep(100);
+	TC_PRINT("[Debug]: before clear cb_cnt\n");
 	cb_cnt = 0;
-	gpio_pin_write(gpio_dev, PIN_OUT,
-		       (polarity == AIO_CMP_POL_RISE) ? 1 : 0);
+	TC_PRINT("[Debug]: after clear cb_cnt\n");
```

Comment:
what are these debugs for?
cb_cnt is just a static int..
unless it gets updated in an interrupt handler or something....in which case make it volatile